src/core/init/stream_events_emitter/get_scheduled_events.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import config from "../../../config";
+import Manifest from "../../../manifest";
+import areSameStreamEvents from "./are_same_stream_events";
+import { IStreamEventPrivateData } from "./types";
+
+const { STREAM_EVENT_EMITTER_POLL_INTERVAL } = config;
+const boundsShift = (STREAM_EVENT_EMITTER_POLL_INTERVAL * 0.6) / 1000;
+
+/**
+ * Refresh local scheduled events list
+ */
+function getScheduledEvents(currentScheduledEvents: IStreamEventPrivateData[],
+                            manifest: Manifest): IStreamEventPrivateData[] {
+  const scheduledEvents: IStreamEventPrivateData[] = [];
+  const { periods } = manifest;
+  for (let i = 0; i < periods.length; i++) {
+    const period = periods[i];
+    const { streamEvents } = period;
+    if (streamEvents !== undefined) {
+      streamEvents.forEach((event) => {
+        const eventStart = (event.presentationTime ?? 0) / event.timescale;
+        const start = period.start + eventStart;
```

Comment:
OK, i found that weird.

I would immediately assume that event.presentationTime would be the absolute time. Isn't it un-natural that way ?

For example, `getSegments`, `getLastPosition` and `getFirstPosition` in a `RepresentationIndex` return the absolute time,